src/inmanta/data/__init__.py

```diff
@@ -4501,6 +4504,40 @@ async def trim(cls, environment: UUID, connection: Optional[Connection] = None)
             connection=connection,
         )
 
+    @classmethod
+    async def mark_orphans(
```

Comment:
Any reason to calculate the orphans rather than to pass them from the scheduler? It should have them in memory already, shouldn't it?